lib/iris/fileformats/netcdf.py

```diff
@@ -1994,77 +1885,97 @@ def _create_cf_coord_variable(self, cube, dimension_names, coord):
+                for index in np.ndindex(orig_shape):
                     index_slice = tuple(list(index) + [slice(None, None)])
-                    cf_var[index_slice] = list(
-                        "%- *s" % (string_dimension_depth, coord.points[index])
+                    new_data[index_slice] = list(
+                        "%- *s" % (string_dimension_depth, data[index])
                     )
+                data = new_data
         else:
-            # Identify the collection of coordinates that represent CF-netCDF
-            # coordinate variables.
-            cf_coordinates = cube.dim_coords
+            # A normal (numeric) variable.
+            # ensure a valid datatype for the file format.
+            element_type = self._ELEMENT_TYPE_NAMES[type(element)]
+            data = self._ensure_valid_dtype(data, element_type, element)
+
+            # Check if this is a dim-coord.
+            is_dimcoord = element in cube.dim_coords
+
+            if isinstance(element, iris.coords.CellMeasure):
+                # Disallow saving of *masked* cell measures.
+                # NOTE: currently, this is the only functional difference in
+                # variable creation between an ancillary and a cell measure.
+                if ma.is_masked(data):
+                    # We can't save masked points properly, as we don't maintain
+                    # a fill_value.  (Load will not record one, either).
+                    msg = "Cell measures with missing data are not supported."
+                    raise ValueError(msg)
 
-            if coord in cf_coordinates:
+            if is_dimcoord:
                 # By definition of a CF-netCDF coordinate variable this
                 # coordinate must be 1-D and the name of the CF-netCDF variable
                 # must be the same as its dimension name.
-                cf_name = cf_dimensions[0]
-
-            # Get the values in a form which is valid for the file format.
-            points = self._ensure_valid_dtype(
-                coord.points, "coordinate", coord
-            )
+                cf_name = element_dims[0]
 
             # Create the CF-netCDF variable.
             cf_var = self._dataset.createVariable(
-                cf_name, points.dtype.newbyteorder("="), cf_dimensions
+                cf_name, data.dtype.newbyteorder("="), element_dims
             )
 
             # Add the axis attribute for spatio-temporal CF-netCDF coordinates.
-            if coord in cf_coordinates:
-                axis = iris.util.guess_coord_axis(coord)
+            if is_dimcoord:
+                axis = iris.util.guess_coord_axis(element)
                 if axis is not None and axis.lower() in SPATIO_TEMPORAL_AXES:
                     _setncattr(cf_var, "axis", axis.upper())
 
             # Add the data to the CF-netCDF variable.
-            cf_var[:] = points
+            cf_var[:] = data  # TODO: support dask streaming
```

Comment:
@pp-mo This data is being written twice :scream: 

See `line +1951`... right?

Reply:
If you agree, then your `TODO` applies to `line +1951` instead

Reply:
:+1:  will fix this